### ISSUE TITLE
ArduCAM_Host_V2.exe + BMP

### ISSUE THREAD

**Issue by bakerbrown** [ArduCAM/Arduino#423]
When selecting the SetToBMP you get a very psychedelic output that is flipped vertically.

**Comment by bakerbrown** [comment]
[files.zip](https://github.com/ArduCAM/Arduino/files/3367974/files.zip)

**Comment by UCTRONICS** [comment]
@bakerbrown 
Do you use the newest source code and  host v2?

**Comment by bakerbrown** [comment]
@UCTRONICS Yes, am I reporting the same issue? https://www.arducam.com/rgb565-format-issues/

**Comment by UCTRONICS** [comment]
@bakerbrown 
Maybe, are you using https://github.com/ArduCAM/Arduino/blob/master/ArduCAM/examples/mini/ArduCAM_Mini_2MP_Plus_VideoStreaming/ArduCAM_Mini_2MP_Plus_VideoStreaming.ino demo ?

**Comment by bakerbrown** [comment]
Have tried:
ArduCAM_Mini_2MP_OV2640_functions
ArduCAM_Mini_2MP_Plus_functions
both output the same corrupt BMP images! Think it more likely that the error is within ArduCAM_Host_V2.exe Where can I find the source to that application?

**Comment by UCTRONICS** [comment]
@bakerbrown 
We don't provide the application source code. Please attach me  your hardware setup and I will help your camera version. If you are using the old version, you should delete the first data when reading the image data.

**Comment by bakerbrown** [comment]
@UCTRONICS Thanks for your quick responses. "We don't provide the application source code." Sorry the licence is missing, so I assumed it was released under Qt's GPL, LGPL not the Commercial license model?

The camera is Mini-2MP-Plus rev A

**Comment by bakerbrown** [comment]
After more experimentation you can almost recover the image by inserting a NULL after the header. I would strongly advise that the error is in the application that saves the .BMP file and not in the driver.

**Comment by UCTRONICS** [comment]
@bakerbrown 
Thanks for your advise, for our newest version, it does not need to add or delete a byte data. we have optimized our hardware.

**Comment by bakerbrown** [comment]
@UCTRONICS which "newest version"? I have only just bought the hardware and am looking to purchase another 20+ units for the current project, this will not happen if I am unable to resolve this issue.

Please advise which hardware version should I be purchasing, where should I purchase it from?

**Comment by UCTRONICS** [comment]
@bakerbrown 
The version you are using is the newest verson Mini 2MP Plus Rev A. Tomorrow, our engineer will test it again and reply you as soon as possible.

**Comment by ArducamTeam** [comment]
@bakerbrown 
Hello,we tested two examples with Arduino UNO and Mini 2MP PLUS.
Test examples: ArduCAM_Mini_2MP_Plus_VideoStreaming\ArduCAM_Mini_2MP_Plus_functions.
The two examples are perfect for both jpeg shooting on host v2 and BMP viewing, except the BMP image is rotated 180°.Below are two screenshots of my test.
![1](https://user-images.githubusercontent.com/48546045/61024876-88d2db80-a3e1-11e9-9b41-3d0ed22632b5.png)
![2](https://user-images.githubusercontent.com/48546045/61024890-8ff9e980-a3e1-11e9-89c0-8212d92706bc.png)
